src/main/java/skype/home_pages/HomePage.java

```diff
@@ -0,0 +1,308 @@
+                {
+                    System.out.println("New contact is found. You can add the contact to contact list");
+                }
+            }
+        } catch (Exception e) {
+            System.out.println("Contact is not new or it is non existing contact");
+        }
+        webDriverWait.until(ExpectedConditions.visibilityOf(clearSearchBtn));
+        actions.moveToElement(clearSearchBtn).click().build().perform();
+        webDriverWait.until(elementNotVisibleCustom(clearSearchBtn));
+        click(side);
+
+//        try {
+//            if ((existingFoundContact.isDisplayed() & (searchSkypeDirectoryBtn2.isDisplayed()))) {
+//                actions.moveToElement(searchSkypeDirectoryBtn2).click().build().perform();
+//                //webDriverWait.until(ExpectedConditions.visibilityOf(noResultsFoundDirectory));
+//                Thread.sleep(3000);
+//                if (existingFoundContact.isDisplayed()) {
+//                    System.out.println("This contact is in your contact list.");
+//                }
+//
+//            }
+//        } catch (Exception e) {
+//            System.out.println("exception2");
+//        }
+
+
+//            }  else {
+//                    if (noResultsFoundPeople.isDisplayed()& searchSkypeDirectoryBtn2.isDisplayed()) {
+//                        actions.moveToElement(searchSkypeDirectoryBtn2).click().build().perform();
+//                    }
+//                    if (noResultsFoundDirectory.isDisplayed()) {
+//                        System.out.println("No results found. Please, refine your search");
+//                    }
+//                }
+//            }
+//
+//
+//        } catch (Exception e) {
+//            System.out.println("Exception");
+//        }
+//        Thread.sleep(3000);
+
+//        click(newFoundContact);
+//        click(side);
+
+//
+//        if (driver.findElement(By.xpath("//li[@class='grid container']/p[.='No results found']")).getText() == "No results found"){
+//            click(searchSkypeDirectoryBtn);
+//        }  else {
+//            System.out.println("Contact is already in contact list");
+//        }
+
+//        try {
+//            if (noResultsFound.getText() == "No results found"){
+//                Thread.sleep(1000);
+//                actions.moveToElement(searchSkypeDirectoryBtn2).click().build().perform();
+//            }  else {
+//            System.out.println("Contact is already in contact list");
+//            }
+//        }catch (Exception e){
+//            System.out.println("Exception");
+//        }
+    }
+
+    public void searchForExistingNewContact2(String text) throws InterruptedException {
+        waitForElementToBeDisplayed(searchSkypeField);
+        click(searchSkypeField);
+        Actions actions = new Actions(driver);
+        actions.moveToElement(focusedSearchSkypeField).click().sendKeys(text).build().perform();
+        WebDriverWait webDriverWait = new WebDriverWait(driver, 60);
```

Comment:
These two lines are used a lot, could be moved to some utility method, and be invoked like ```waitForElementToBeDisplayed(searchSkypeField);```.